doc/contributing.md

```diff
@@ -16,37 +16,217 @@ and we encourage all to read it carefully.
+    explaining the *reason* behind non-obvious things.
+  - Write tests for the code you wrote/modified if needed.
+    Please refer to [Testing your code](#testing-your-code) or
+    [Testing plots](#testing-plots).
+  - Include an example of new features in the gallery or tutorials.
+    Please refer to [Gallery plots](#gallery-plots) or [Tutorials](#tutorials).
+* Have a good coding style
+  - Use readable code, as it is better than clever code (even with comments).
+  - Follow the [PEP8](http://pep8.org) style guide for code and the
+    [numpy style guide](https://numpydoc.readthedocs.io/en/latest/format.html)
+    for docstrings. Please refer to [Code style](#code-style).
+
+Pull requests will automatically have tests run by GitHub Actions.
+This includes running both the unit tests as well as code linters.
+GitHub will show the status of these checks on the pull request.
+Try to get them all passing (green).
+If you have any trouble, leave a comment in the PR or
+[get in touch](#how-can-i-talk-to-you).
+
+## Contributing Documentation
+
+### PyGMT Documentation Overview
+
+There are four main components to PyGMT's documentation:
+
+* Gallery examples, with source code in Python `*.py` files under the
+  `examples/gallery/` folder.
+* Tutorial examples, with source code in Python `*.py` files under the
+  `examples/tutorials/` folder.
+* API documentation, with source code in the docstrings in Python `*.py`
+  files under the `pygmt/src/` and `/pygmt/datasets/` folders.
+* Getting started/developer documentation, with source text in ReST `*.rst`
+  and markdown `*.md` files under the `doc/` folder.
+
+The documentation are written primarily in
+[reStructuredText](https://docutils.sourceforge.io/rst.html) and built by
+[Sphinx](http://www.sphinx-doc.org/). Please refer to
+[reStructuredText Cheatsheet](https://docs.generic-mapping-tools.org/latest/rst-cheatsheet.html)
+if you are new to reStructuredText. When contributing documentation, be sure to
+follow the general guidelines in the [pull request workflow](#pull-request-workflow)
+section.
+
+There are two primary ways to edit the PyGMT documentation. For simple documentation
+changes, you can easily
+[edit the documentation on GitHub](#editing-the-documentation-on-github). This
+only requires you to have a GitHub account. For more complicated changes, you
+can [edit the documentation locally](#editing-the-documentation-locally). In
+order to build the documentation locally, you first need to
+[set up your environment](#setting-up-your-environment).
+
+### Editing the Documentation on GitHub
 
 If you're browsing the documentation and notice a typo or something that could be
 improved, please consider letting us know by [creating an issue](#reporting-a-bug) or
-submitting a fix.
+(even better) submitting a fix.
 
 You can submit fixes to the documentation pages completely online without having to
 download and install anything:
```

Comment:
This workflow is great, but I'm wondering if we could use an ordered list to be a little more friendly to new Git users. See how it looks:

ordered | unordered
--- | ---
![](https://user-images.githubusercontent.com/50591376/122664150-217b5400-d1d2-11eb-87bd-982b1f49718b.png) | ![](https://user-images.githubusercontent.com/50591376/122664154-24764480-d1d2-11eb-977d-185a30dbe263.png)